### ISSUE TITLE
ADVANCED_OPTIMIZATION renaming appears to behave inconsistently

### ISSUE THREAD

**Issue by froodian** [google/closure-compiler#1019]
This code

```
var TestObject = function(message, duration, dismissType) {
  this['message'] = message;
  this['duration'] = duration;
  this['dismissType'] = dismissType;
};
var DismissTypes = {
  "AUTO_DISMISS": 'AUTO_DISMISS',
  "MANUAL": 'SWIPE'
};

var instance = new TestObject('my message', 1000, DismissTypes['AUTO_DISMISS']);

console.log(instance.message);
console.log(instance.duration);
console.log(instance.dismissType);
```

produces this compiled output:

```
var a=new function(b,c,d){this.message=b;this.duration=c;this.dismissType=d}("my message",1E3,"AUTO_DISMISS");console.log(a.message);console.log(a.duration);console.log(a.a);
```

why are message and duration preserved, but dismissType is not?  Is this correct?  I was unable to find an explanation for this behavior in the documentation.

**Comment by ChadKillingsworth** [comment]
It's covered [in the FAQ](https://github.com/google/closure-compiler/wiki/FAQ#some-of-my-properties-are-getting-renamed-but-some-arent-why)

TL;DR `message` and `duration` are probably properties in externs, where `dismissType` is not.
